pytorch_lightning/core/lightning.py

```diff
@@ -1088,77 +1088,26 @@ def val_dataloader(self):
     @classmethod
     def load_from_metrics(cls, weights_path, tags_csv, map_location=None):
         r"""
-        You should use `load_from_checkpoint` instead!
-        However, if your .ckpt weights don't have the hyperparameters saved, use this method  to pass
-        in a .csv with the hparams you'd like to use. These will  be converted  into a argparse.Namespace
-        and passed into  your LightningModule for use.
-
-        Args:
-
-            weights_path (str): Path to a PyTorch checkpoint
-            tags_csv (str): Path to a .csv with two columns (key, value) as in this
-
-            Example::
-                key,value
-                drop_prob,0.2
-                batch_size,32
-
-            map_location (dict | str | torch.device | function):
-                If your checkpoint saved a GPU model and you now load on CPUs
-                or a different number of GPUs, use this to map to the new setup
-                (example: {'cuda:1':'cuda:0'}).
-                The behaviour is the same as in
-                `torch.load <https://pytorch.org/docs/stable/torch.html#torch.load>`_.
-
-        Return:
-            LightningModule with loaded weights and hyperparameters (if available).
-
-        Example
-        -------
-        .. code-block:: python
-
-            pretrained_model = MyLightningModule.load_from_metrics(
-                weights_path='/path/to/pytorch_checkpoint.ckpt',
-                tags_csv='/path/to/hparams_file.csv',
-                on_gpu=True,
-                map_location=None
-            )
-
-            # predict
-            pretrained_model.eval()
-            pretrained_model.freeze()
-            y_hat = pretrained_model(x)
+        Warning:
+            Deprecated in version 0.6.1.
+            You should use `load_from_checkpoint` instead.
+            Will be removed in v0.8.0.
         """
-
-        hparams = load_hparams_from_tags_csv(tags_csv)
-        hparams.__setattr__('on_gpu', False)
-
-        if map_location is not None:
-            checkpoint = torch.load(weights_path, map_location=map_location)
-        else:
-            checkpoint = torch.load(weights_path, map_location=lambda storage, loc: storage)
-
-        # add the hparams from csv file to checkpoint
-        checkpoint['hparams'] = vars(hparams)
-
-        model = cls._load_model_state(checkpoint)
-        return model
+        return cls.load_from_checkpoint(weights_path, tags_csv=tags_csv, map_location=map_location)
 
     @classmethod
-    def load_from_checkpoint(cls, checkpoint_path, map_location=None):
+    def load_from_checkpoint(cls, checkpoint_path, tags_csv=None, map_location=None):
```

Comment:
```suggestion
    def load_from_checkpoint(cls, checkpoint_path: str, tags_csv: str = None, map_location:Union[dict, str, torch.device, Callable] = None):
```
than add `from typing import Union, Callable` and remove types in docs

Reply:
we shall rather add the new parameter at the and some pas usage may use potion argument and it will crash

Reply:
the typing I have already done in #946 and I will rebase depending which will be merged first.

good point about the breaking of positional arguments. will move it to the end.

Reply:
sure but just for luck of collision, I would recommend to do it also here and remove from docstring...

Reply:
what? that doesn't make sense. the point of a rebase is to resolve the merge conflicts. you are basically asking me to copy the code from one branch to the next manually... that's not how git works as far as I know. Even though it is a small change you are requesting, it is logically not the right thing to do. Imagine the case when a PR/commit needs to be reverted for some reason. Then it would be much harder to undo a particular error when a PR/commit does multiple things at once. I highly advise against introducing this noise to PRs. Please reconsider the request.

Reply:
Git doesn't understand moving code blocks so if you moved part of docs to new function ist just delete and new code so if this is merged before #946 there won't be any automatic consequent change in the later PR, you have to remember and make it automatically...
I understand that copy changes is not correct solution, but each PR shall be complete... So you say that this one is depended on #946 so mark it in title that it is blocked by or still wip
Is this fine for you?

Reply:
I added the types as you originally requested.
Btw the merge tools these days are pretty good actually, I never had a concern like this one :)

Reply:
Unfortunately we had in my company and it quite annoying... So sorry for being so irritating =)

Reply:
ok no problem, I understand now, thanks. it should be good now and not give conflicts in the other PR.